services/application-binding.html.md.erb

```diff
@@ -25,6 +27,140 @@ $ cf restart my-app
+There are the following three delivery methods, which are mutually exclusive. The developer can choose which one to use and needs to adjust the consumption in the app accordingly.
+For details about consuming credentials specific to your development framework, see the Service Binding section in the documentation for your framework's [buildpack](../../buildpacks/index.html).
+
+#### <a id='vcap-services'></a> Default VCAP_SERVICES environment variable
+
+By default credentials and additional metadata for all bound service instances are stored in the <a href="../deploy-apps/environment-variable.html">VCAP_SERVICES</a> environment variable.
+
+There are two methods developers can use to have their apps consume binding credentials.
+
+* **Parse the JSON yourself:** See the documentation for [VCAP_SERVICES](../deploy-apps/environment-variable.html#VCAP-SERVICES). Helper libraries are available for some frameworks.
+* **Auto-configuration:** Some buildpacks create a service connection for you by creating additional environment variables, updating config files, or passing system parameters to the JVM.
+
+For checking the binding data manually app devlopers can use `cf env`. See [View Env](../deploy-apps/environment-variable.html#view-env).
+
+<p class="note important">
+The maximum size of content of the VCAP_SERVICES environment variable is 130 KB. If you want to bind more services to the app or the services you want to bind provide lots of data, you can use one of the other methods.</p>
+
+#### <a id='file-based-vcap-services'></a> File-based VCAP services (experimental)
+
+This delivery method is used when the app feature [file-based-vcap-services](https://v3-apidocs.cloudfoundry.org/index.html#app-features) is enabled.
+The environment variable [VCAP_SERVICES_FILE_PATH](../deploy-apps/environment-variable.html#VCAP-SERVICES-FILE-PATH) will be made available in the app container and contains the path to a file containing the same content as the VCAP_SERVICES environment variable above.
+
+The binding data can be viewed like this:
+
+~~~
+cf ssh <app_name> -c 'cat $VCAP_SERVICES_FILE_PATH'
+~~~
+
+<p class="note important">
+The vcap_services file content cannot exceed 1 MB, otherwise an `IncompatibleBindings` error is raised.</p>
+
+#### <a id='service-binding-k8s'></a> Service Binding K8s (experimental)
+
+This delivery method is used when the app feature [service-binding-k8s](https://v3-apidocs.cloudfoundry.org/index.html#app-features) is enabled.
+The environment variable [SERVICE_BINDING_ROOT](../deploy-apps/environment-variable.html#SERVICE-BINDING-ROOT) will be made available in the app container and contains the path to the root folder of a file structure containing binding information.
+This method is fully compatible with the specification on [servicebinding.io](https://servicebinding.io/), which is used by Kubernetes as well.
+
+The binding data can be viewed like this:
+
+~~~
```

Comment:
3 backticks is the usual way we delimit code snippets. I haven't seen this before, so I don't know if it will work.
This comment applies to all of the new code samples in this file.